lib/components/map/enhanced-stop-marker.js

```diff
@@ -0,0 +1,116 @@
+import { styled as BaseMapStyled } from '@opentripplanner/base-map'
+import coreUtils from '@opentripplanner/core-utils'
+import FromToLocationPicker from '@opentripplanner/from-to-location-picker'
+import { divIcon } from 'leaflet'
+import React, { Component } from 'react'
+import ReactDOMServer from 'react-dom/server'
+import { connect } from 'react-redux'
+import { Marker, Popup } from 'react-leaflet'
+import styled from 'styled-components'
+
+import * as mapActions from '../../actions/map'
+import * as uiActions from '../../actions/ui'
+import { getStopName } from '../../util/viewer'
+
+const { getTextWidth } = coreUtils.itinerary
+
+const BaseStopIcon = styled.div`
+  background: #fff;
+  border: ${props => props.active ? '#000 3px' : '#333 1px'} solid;
+  border-radius: 17px;
+  color: #000;
+  font-size: 16px;
+  font-weight: bold;
+  height: 12px;
+  line-height: 0px;
+  padding-left: 7px;
+  padding-top: 12px;
+  width: ${props => `${props.width || 12}px`}
+`
+
+class EnhancedStopMarker extends Component {
+  onClickView = () => {
+    const { setViewedStop, stop } = this.props
+    setViewedStop({ stopId: stop.id })
+  }
+
+  onFromClick = () => {
+    this.setLocation('from')
+  }
+
+  onToClick = () => {
+    this.setLocation('to')
+  }
+
+  setLocation (locationType) {
+    const { setLocation, stop } = this.props
+    const { lat, lon, name } = stop
+    setLocation({ location: { lat, lon, name }, locationType })
+  }
+
+  render () {
+    const { activeStopId, languageConfig, stop } = this.props
+    const { code, id, lat, lon, name } = stop
+    const [, stopId] = id.split(':')
+    const stopCode = code || stopId
+    const routeShortNames = stop.routes?.map(r => r.shortName).join(' ')
+    const stopLabel = routeShortNames || ''
```

Comment:
I'm not a huge fan of how this label looks currently. It's a little odd to see this label with a bunch of numbers. It might be ok for someone with deep knowledge of the area, but in crowded areas with lots of routes it doesn't look great. Furthermore, the fact that OTP treats stops from different feeds as completely different stops is not helping the situation either.

<img width="1274" alt="Screen Shot 2021-08-03 at 2 45 59 PM" src="https://user-images.githubusercontent.com/3112493/128090664-811a9936-d38b-4fd9-9cb2-1dc7293a1efe.png">

I would've expected the name of the stop to be displayed somewhere and maybe just that. Perhaps transitive can actually help out here by rendering routes with labels and rendering labels in a way that they don't overlap. The stops from different feeds thing would still be a challenging item to resolve, but that the resolution there is a back-end solution in OTP or merging the data that goes into OTP.

Reply:
I think we need to discuss this item together with Jon.